packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/nodes/NodeSizeDetailsTests.java

```diff
@@ -20,17 +26,147 @@ public class NodeSizeDetailsTests extends RestTest
     private UserModel user1;
     private SiteModel siteModel;
     private FolderModel folder;
+    private String jobId;
 
     @BeforeClass(alwaysRun = true)
     public void dataPreparation()
     {
-        user1 = dataUser.createRandomTestUser("User-1");
+        user1 = dataUser.getAdminUser();
         siteModel = dataSite.usingUser(user1).createPublicRandomSite();
         folder = dataContent.usingUser(user1).usingSite(siteModel).createFolder(FolderModel.getRandomFolderModel());
     }
 
     /**
-     *
+     * calculateNodeSizeForSingleFile testcase
+     */
+
+    @TestRail(section = {TestGroup.REST_API, TestGroup.NODES}, executionType = ExecutionType.SANITY)
+    @Test(groups = {TestGroup.REST_API, TestGroup.NODES, TestGroup.SANITY})
+    public void calculateNodeSizeForSingleFile() throws Exception
+    {
+
+        STEP("1. Create a folder in the test site.");
+        folder = dataContent.usingUser(user1).usingSite(siteModel).createFolder(FolderModel.getRandomFolderModel());
+
+        STEP("2. Upload a text document to the folder.");
+        long fileSize;
+
+        restClient.authenticateUser(user1).configureRequestSpec().addMultiPart("filedata", Utility.getResourceTestDataFile("sampleLargeContent.txt"));
+        RestNodeModel fileNode = restClient.withCoreAPI().usingNode(folder).createNode();
+        restClient.assertStatusCodeIs(HttpStatus.CREATED);
+        fileNode.assertThat().field("id").isNotNull()
+                .and().field("name").is("sampleLargeContent.txt")
+                .and().field("content.mimeType").is(FileType.TEXT_PLAIN.mimeType);
+        fileSize = Utility.getResourceTestDataFile("sampleLargeContent.txt").length();
+
+        STEP("3. Wait for 30 seconds so that the content is indexed in Search Service.");
+        Utility.waitToLoopTime(30);
+
+        RestSizeDetailsModel restSizeDetailsModel = restClient.authenticateUser(user1).withCoreAPI().usingNode(folder).executeSizeDetails();
+        restClient.assertStatusCodeIs(HttpStatus.ACCEPTED);
+        restSizeDetailsModel.assertThat().field("jobId").isNotEmpty();
+
+        jobId = restSizeDetailsModel.getJobId();
+
+        STEP("4. Wait for 3 seconds for the processing to complete.");
+        Utility.waitToLoopTime(3);
```

Comment:
Please avoid waiting a fixed time. These test might be run on environment with different resources. Sometimes 3 seconds might be enough sometimes it might be not. It will lead to flaky tests and will require additional maintenance in the future. Try to increase the wait time but during that time check the assertions multiple times. You can do it manually in the loop or you can use `org.awaitility.Awaitility` for instance.

Reply:
Acknowledged.
We have replaced the fixed time with **Awaitility.await().atMost(Duration.ofSeconds())** and configured it to be used when retrieving the size details.